packages/core/src/renderer/Callout.ts

```diff
@@ -0,0 +1,405 @@
+  );
+
+  if (pointInBox(toPt(anchor), linesR)) {
+    console.log("Anchor in box");
+    return ptsR;
+  }
+
+  // callout center -> anchor, parallel to callout direction
+  const vec = ops.vnormalize(ops.vsub(center, anchor));
+  const stemStart = ops.vmove(anchor, calloutEndPadding, vec); // Pointy part
+
+  // stemSeg_i = one side of the speech bubble "tail"
+  // Extrusions from normal on either side of the stem
+  const t = ops.vnorm(ops.vsub(center, stemStart));
+  const stemSide1Start = ops.vmove(
+    stemStart,
+    calloutThickness / 2,
+    ops.rot90(vec)
+  );
+  const stemSide1End = ops.vmove(stemSide1Start, t, vec);
+  const stemSeg1 = [stemSide1Start, stemSide1End];
+
+  const stemSide2Start = ops.vmove(
+    stemStart,
+    calloutThickness / 2,
+    ops.rot90(ops.rot90(ops.rot90(vec)))
+  );
+  const stemSide2End = ops.vmove(stemSide2Start, t, vec);
+  const stemSeg2 = [stemSide2Start, stemSide2End];
+
+  // intersectPts = the places where the extruded line intersects with the rectangle segments, where the stem should be drawn in the polygon
+  let intersectPt1 = undefined;
+  let side1 = undefined;
+
+  if (intersects(segsR.top, stemSeg1)) {
+    intersectPt1 = intersection(segsR.top, stemSeg1);
+    side1 = "top";
+  } else if (intersects(segsR.bot, stemSeg1)) {
+    intersectPt1 = intersection(segsR.bot, stemSeg1);
+    side1 = "bot";
+  } else if (intersects(segsR.left, stemSeg1)) {
+    intersectPt1 = intersection(segsR.left, stemSeg1);
+    side1 = "left";
+  } else if (intersects(segsR.right, stemSeg1)) {
+    intersectPt1 = intersection(segsR.right, stemSeg1);
+    side1 = "right";
+  } else {
+    console.log("no intersection 1", segsR, stemSeg1);
+    // throw Error("no intersection for point 1");
+  }
+
+  let intersectPt2 = undefined;
+  let side2 = undefined;
+
+  if (intersects(segsR.top, stemSeg2)) {
+    intersectPt2 = intersection(segsR.top, stemSeg2);
+    side2 = "top";
+  } else if (intersects(segsR.bot, stemSeg2)) {
+    intersectPt2 = intersection(segsR.bot, stemSeg2);
+    side2 = "bot";
+  } else if (intersects(segsR.left, stemSeg2)) {
+    intersectPt2 = intersection(segsR.left, stemSeg2);
+    side2 = "left";
+  } else if (intersects(segsR.right, stemSeg2)) {
+    intersectPt2 = intersection(segsR.right, stemSeg2);
+    side2 = "right";
+  } else {
+    console.log("no intersection 2", segsR, stemSeg2);
+    // throw Error("no intersection for point 2");
+  }
+
+  const stemPts = [
+    intersectPt1 ? intersectPt1 : stemSide1End,
+    anchor,
+    intersectPt2 ? intersectPt2 : stemSide2End,
+  ];
+
+  // TODO: The corner should be skipped if the points "include" it?
+
+  // [ ] <- rect center to middle of right segment is zero angle
+  const zeroAngleVec = ops.vdiv(
+    ops.vadd(cornersR.bottomRight, cornersR.topRight),
+    2
+  );
+
+  // Sort points by their angle to the center of the callout
+  const ptsSorted = sortBy(ptsR.concat(stemPts), (pt) =>
+    angleBetween2(
```

Comment:
I see the `vectors` dependency is used here (and only here?). Does the library have more vector ops, so you don't have to implement them above?  Currently, this file is a mixture of library and hand-written functions.

Reply:
Yes, the library has more vectors, but I don't use it because (see prev comment). Probably the easier thing to do is to remove the dependency and just implement `angleBetween2`.

Reply:
Yep, that sounds good!